app/models/project.rb

```diff
@@ -158,15 +158,13 @@ def self.search_scope(query)
     visible.like(query)
   end
 
-  def add_member(user, roles)
+  def add_member!(user, roles)
+    raise "Unsafe operation. Use Members::CreateService" unless Rails.env.test?
```

Comment:
I can see why you did this, but it's depremental to my good feeling when the test suite succeeds as it now no longer guarantees in this concrete case to do the same

Reply:
I'll see how many places still used the functionality in the tests. If there aren't too many or if I can rework it via some regexp I'll give it another try.

Reply:
No regexp. But did it by hand.